RavenDB-11919 Fix DistributedRevisionsSubscription test

karmeli87 opened the pull request:
We have to wait for the subscription to be created on the responsible node.

karmeli87 committed:
RavenDB-11919 Fix DistributedRevisionsSubscription test

We have to wait for the subscription to be created on the responsible node.

ppekrol commented on src/Raven.Server/Documents/Handlers/SubscriptionsHandler.cs:
```diff
@@ -152,13 +152,27 @@ public async Task Create()
+                var name = options.Name ?? subscriptionId.ToString();
+
+                // need to wait on the relevant remote node
+                var node = Database.SubscriptionStorage.GetResponsibleNode(name);
```
can we avoid opening 2 server-store transactions by adding ctx argument to this method and moving it next to `WaitForExecutionOnSpecificNode` ?

ppekrol replied:
Done

ppekrol committed:
RavenDB-11919
- passing serverContext to GetResponsibleNode to avoid opening multiple serverStore read transactions
- fixed conflicts

ayende commented on src/Raven.Server/Web/RequestHandler.cs:
```diff
@@ -132,6 +138,63 @@ private static bool HeadersAllowGzip(IDictionary<string, StringValues> headers,
 
+        protected async Task WaitForExecutionOnSpecificNode(TransactionOperationContext context, ClusterTopology clusterTopology, string node, long index)
+        {
+            using (var requester = ClusterRequestExecutor.CreateForSingleNode(clusterTopology.GetUrlFromTag(node), ServerStore.Server.Certificate.Certificate))
```
Why is this an instance method that also create a new _instance_ for this?
What does this give us?

ayende commented on src/Raven.Server/Web/RequestHandler.cs:
```diff
@@ -132,6 +138,63 @@ private static bool HeadersAllowGzip(IDictionary<string, StringValues> headers,
+                    var url = member == ServerStore.NodeTag ?
+                        Server.WebUrl :
+                        clusterTopology.GetUrlFromTag(member);
+                    var requester = ClusterRequestExecutor.CreateForSingleNode(url, ServerStore.Server.Certificate.Certificate);
```
Same thing here, actually. We are creating new instances from an instance method, which seems strange.
